### DISCUSSION TITLE
Project on update/create (set values from another object en masse)?

### DISCUSSION THREAD

**Discussion by Metadorius** [linq2db/linq2db#3808] [Q&A] [answered]
In my application I tried to use entity-DTO mapping using `Expression<Func<Entity, DTO>>` and vice versa like described here: https://github.com/linq2db/linq2db/issues/1283#issuecomment-413509043
This works great for projecting using a select, but what do I do when I need to update/insert a new record? I've skimmed over Update and Set extension methods but couldn't find anything. Ideally I would want something like this:

```cs

            Expression<Func<SomeDTO, SomeEntityTable>> projectExpr =
                x => new SomeEntity
                {
                    ID = x.ID,
                    Name = x.Name,
                    // ...
                }; // this is just so that I can write two mapping expressions per DTO and don't ever repeat them, for stuff like CRUD
            
            // ...

            using var db = ConnectionFactory.Instance.GetMainDB();
            await db.SomeEntityTable
                .Where(e => e.ID == dto.ID)
                .Set(dto, projectExpr) // dto is of SomeDTO type here; this will set ONLY the values that are written in the expression
                .Set(e => e.LastEditedAt, DateTime.Now()) // able to append some more stuff after 
                .UpdateAsync();
```

I suspect this might be pretty easy to implement on my side, but unfortunately I lack knowledge on expression construction for IQueryable :( Can anyone help me write such a method?

**Comment by Metadorius** [comment]
Still in need of a solution on this one 🙂

**Reply by sdanyliv** [reply]
To keep knowledge sharing better ask this question on StackOverflow. I will prepare answer.

**Reply by Metadorius** [reply]
> To keep knowledge sharing better ask this question on StackOverflow. I will prepare answer.

Done, and thanks for taking your time on this 💯 
https://stackoverflow.com/questions/74786123/project-on-update-create-set-values-from-another-object-en-masse-in-linq2db

**Reply by sdanyliv** [reply]
Answered.

**Reply by sdanyliv** [reply]
Please mark SO answer as accepted if answer is OK.

**Reply by Metadorius** [reply]
Thank you very much, done! yeah I just quickly checked it yesterday after work so didn't yet do anything else 😄 BTW, can I support you somehow?

**Comment by Metadorius** [comment, marked as answer]
Answer posted by @sdanyliv on Stack Overflow:
https://stackoverflow.com/a/74788323/20766970
